### PR TITLE
add xgdp

### PR DESCRIPTION
**Pull request by tereshchuk1**
Adds **XGDP**. The model combines a GNN branch for drug molecular graphs with a 1D-CNN branch for cell line gene expression, with optional cross-attention.

Adapted from the [original repository](https://github.com/SCSE-Biomedical-Computing-Group/XGDP) with the following changes for drevalpy integration:
- Merged `upstream/development` and resolved conflicts
- Edge features aligned to 7-dim to match the existing `create_drug_graphs.py` featurizer
- CNN input size computed dynamically instead of hardcoded
- Fixed MSE loss shape mismatch
- Added `ensure_drug_graphs` fixture to `tests/conftest.py`
- Suppressed `S614` for `torch.load(..., weights_only=False)` (required for `torch_geometric.data.Data`)

### CODE REVIEW

File: drevalpy/models/XGDP/xgdp.py [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+        loggers = []
+        if self.wandb_project is not None:
+            from pytorch_lightning.loggers import WandbLogger
+
+            logger = WandbLogger(project=self.wandb_project, log_model=False)
+            loggers.append(logger)
+
+        trainer = pl.Trainer(
+            # max_epochs=self.hyperparameters.get("epochs", 100), #changed to 10 fro testing
+            max_epochs=1,
```
Comment (PascalIversen) [inline drevalpy/models/XGDP/xgdp.py]: max_epoch=1 is hardcoded here, it doesnt use the epochs from the hpams

File: drevalpy/models/XGDP/xgdp.py [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+    drug_views = ["drug_graph"]
+
+    def __init__(self) -> None:
+        """Initialize the XGDP model."""
+        super().__init__()
+        self.DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        self.model: XGDPModule | None = None
+        self.hyperparameters: dict[str, Any] = {}
+        self.gene_expression_scaler: StandardScaler | None = None
+        self.gene_expression_normalizer: MinMaxScaler | None = None
```
Comment (PascalIversen) [inline drevalpy/models/XGDP/xgdp.py]: the scaler and normalizer are declared here but never fit or applied, so the gene expression goes into the cnn unscaled. how does the publication do it?

Comment (tereshchuk1) [reply drevalpy/models/XGDP/xgdp.py]: Checked the original XGDP repo - gene expression is never scaled/normalized there, only variance basedfiltering is applied before it goes into the CNN. So this is currently dead code. Should I remove it?
